snuba/query/query.py

```diff
@@ -123,6 +124,15 @@ def add_conditions(
     ) -> None:
         self.__extend_sequence("conditions", conditions)
 
+    def get_prewhere(self) -> Sequence[Condition]:
+        return self.__prewhere_conditions
+
+    def set_prewhere(
+        self,
+        conditions: Sequence[Condition]
+    ) -> None:
+        self.__prewhere_conditions = conditions
```

Comment:
I want to get back on this and revisit the decision of adding a PREWHERE field to the Snuba query later when I will be working on Clickhouse formatting. Prewhere fields could be processed at that point since PREWHERE is something Clickhouse specific and all the information needed is in the data source, which is accessible from the query. 
Though this means the formatting logic becomes more complex.

Reply:
_Do you have any sense of how much work it would be to move it? _

I think it is complex enough to do it in its own PR, the only reason is that it would change the logic of the prewhere processor itself. Now (and before this PR) the prewhere processing updated the query: it changes the set of conditions. If we do that in the format, we would not update the query object anymore, other wise we go back potentially to that state where formatting the query made changes to it.

So I would keep it separate since this PR does not touch the PREWHERE processing logic.